src/utils/handleOptions.js

```diff
@@ -0,0 +1,232 @@
+            errorMessages.enum(modelName, key, userValue, enums, defaultValue)
+          );
+          filteredObject[key] = defaultValue;
+        } else {
+          filteredObject[key] = checkedValue;
+        }
+      }
+      // If the type is a number, apply the min and max rules
+      else if (type === "number") {
+        const min = evaluate(filteredObject, rules.min) ?? -Infinity;
+        const max = evaluate(filteredObject, rules.max) ?? Infinity;
+        const integer = evaluate(filteredObject, rules.integer) ?? false;
+        const multipleOf = evaluate(filteredObject, rules.multipleOf);
+        const checkedValue = isInRange(userValue, min, max);
+
+        if (integer && !Number.isInteger(userValue)) {
+          console.warn(
+            errorMessages.numberInteger(modelName, key, userValue, defaultValue)
+          );
+          filteredObject[key] = defaultValue;
+        } else if (multipleOf !== undefined && userValue % multipleOf !== 0) {
+          console.warn(
+            errorMessages.numberMultipleOf(
+              modelName,
+              key,
+              userValue,
+              multipleOf,
+              defaultValue
+            )
+          );
+          filteredObject[key] = defaultValue;
+        } else if (checkedValue === undefined) {
+          console.warn(
+            errorMessages.numberRange(
+              modelName,
+              key,
+              userValue,
+              min,
+              max,
+              defaultValue
+            )
+          );
+          filteredObject[key] = defaultValue;
+        } else {
+          filteredObject[key] = checkedValue;
+        }
+      }
+      // If type is boolean, string, or object, use the user value as is
+      else if (type === "boolean" || type === "string" || type === "object") {
+        filteredObject[key] = userValue;
+      }
+      // Throw an error if the type in moldObject is not "enum", "number", "boolean", "string", or "object"
+      // This should never happen if the moldObject is correctly defined.
+      else {
+        throw new Error(`Unknown type "${type}" for ${key} in moldObject.`);
+      }
+    }
+  }
+  console.log(filteredObject);
+  return filteredObject;
+}
+
+/**
+ * A function that takes a user value and checks if it is a valid underlying model name.
+ * Returns the model name the user provided if it is valid, otherwise returns the default model name.
+ *
+ * @param {string} userValue - the value provided by the user
+ * @param {string[]} possibleValues - the available model names
+ * @param {string} defaultValue  - the default model to use if the userValue is not valid
+ * @param {string} ml5ModelName - the name of the ml5 model
+ *
+ * @returns {string} - the model name to use
+ */
+function handleModelName(
+  userValue,
+  possibleValues,
+  defaultValue,
+  ml5ModelName
+) {
+  const modelName = checkEnum(userValue, possibleValues, true);
+  if (userValue === undefined) {
+    return defaultValue;
+  }
```

Comment:
It makes more sense if you swap the order of these two, as there's no need to call `checkEnum` if you're just returning the default.